# Prevent namespace import declarations. (no-namespace-imports)

Prior to TypeScript 2.7, CommonJS/AMD/UMD moduels were treated in the same way as ES6 modules. This was inconsistent with ECMAScript spec and resulted in namespace imports, such as `import * as foo from "foo"`, for a CommonJS/AMD/UMD module as equivalent to `const foo = require("foo")` (where `foo` can be more than just a non-callable object). TypeScript 2.7 and later adds `--esModuleInterop` which will change the generated output from TypeScript closer to that generated by Babel where namespace imports are properly flagged as uncallable and default imports work as expected.

## Rule Details

This rule will report and change namespace imports to default imports on a pre-defined, but configurable, list of modules.

Examples of **incorrect** code for this rule:

```ts

import * as React from 'react';

```

Examples of **correct** code for this rule:

```ts

import React from 'react';

```


### `modules`

```json
{
  "shopify/no-namespace-imports": [
    "error",
    {
      "modules": ["react", "some-custom-module"]
    }
  ]
}
```

This array options adds modules to the pre-defined list and this rule will report and fix their namespaced imports.

### `ignore`

```json
{
  "shopify/no-namespace-imports": [
    "error",
    {
      "ignore": ["react", "react-dom"]
    }
  ]
}
```

This array option whitelists items in the pre-defined `modules` list. Modules defined in this option will not have their namespace imports reported or fixed. There are eight possible values.

* `'@shopify/app-bridge'`
* `'@shopify/app-bridge/actions'`
* `'@shopify/app-bridge/validate/actions/navigation'`
* `'@shopify/app-bridge/validate/actions/resourcePicker'`
* `'@shopify/app-bridge/validate/actions'`
* `'@shopify/react-form-state'`
* `'@shopify/safe-redirect'`
* `'d3'`
* `'faker'`
* `'fs'`
* `'glob'`
* `'history'`
* `'http'`
* `'koa'`
* `'react-dom'`
* `'react-router'`
* `'redux'`
* `'topojson'`
* `'prop-types'`
* `'react'`
* `'url'`


## When Not To Use It

If you are on a Typescript version that does not support the `--esModuleInterop` flag or do not wish to modify your namespace imports, you can safely disable this rule.

## Further Reading

* [Typescript 2.7 Release Notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#support-for-import-d-from-cjs-from-commonjs-modules-with---esmoduleinterop)
